sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuGetJsonObject.scala

```diff
@@ -83,49 +83,22 @@ private[this] object JsonPathParser extends RegexParsers {
     }
   }
 
-  def containsUnsupportedPath(instructions: List[PathInstruction]): Boolean = {
-    // Gpu GetJsonObject is not supported if JSON path contains wildcard [*]
-    // see https://github.com/NVIDIA/spark-rapids/issues/10216
-    instructions.exists {
-      case Wildcard => true
-      case Named(name) if name == "*" => true
-      case _ => false
+  def unzipInstruction(instruction: PathInstruction): (Int, String, Long) = {
+    instruction match {
+      case Subscript => (0, "", -1)
```

Comment:
Use string for types instead of int to improve readability.
In JNI repo, magic integer is hard to understand.

Reply:
Updated to enum now.